worksheet_D/oxo.py

```diff
@@ -0,0 +1,115 @@
+class OxoBoard:
+    def __init__(self):
+        """ The initialiser. Initialise any fields you need here. """
+        self.oxoList = ([0,0,0],[0,0,0],[0,0,0])
+
+    def get_square(self, x, y):
+        """ Return 0, 1 or 2 depending on the contents of the specified square. """
+        return self.oxoList[x][y]
+
+    def set_square(self, x, y, mark):
+        """ If the specified square is currently empty (0), fill it with mark and return True.
+            If the square is not empty, leave it as-is and return False. """
+        if self.oxoList[x][y] == 0:
+            self.oxoList[x][y] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        """ If there are still empty squares on the board, return False.
+            If there are no empty squares, return True. """
+        xPos = 0
+        yPos = 0
+        squaresFilled = True
+
+        while xPos <3:
+            while yPos < 3 :
+                if self.oxoList[xPos][yPos] == 0:
+                    squaresFilled = False
+
+        return squaresFilled
+
+    def get_winner(self):
+        """ If a player has three in a row, return 1 or 2 depending on which player.
+            Otherwise, return 0. """
+
+        xValue = 0
+        while xValue <3:
+
+            if self.oxoList[0][0] == self.oxoList[1][1] and self.oxoList[0][0] == self.oxoList[2][2]:
+                return self.oxoList[0][0]
+            elif self.oxoList[0][2] == self.oxoList[1][1] and self.oxoList[2][0] == self.oxoList[0][2]:
+                return self.oxoList[0][2]
+            elif self.oxoList[xValue][0] == self.oxoList[xValue][1] and self.oxoList[xValue][0] == self.oxoList[xValue][2]:
+                return self.oxoList[xValue][0]
+            elif self.oxoList[0][xValue] == self.oxoList[1][xValue] and self.oxoList[0][xValue] == self.oxoList[2][xValue]:
+                return self.oxoList[0][xValue]
+            else:
+                return 0
```

Comment:
This will cause the function to return after checking only the first row and column -- a win in the second or third row or column will be missed.